lib/types/src/trapcode.rs

```diff
@@ -120,6 +120,18 @@ impl FromStr for TrapCode {
     }
 }
 
+/// After the stack is unwound via asyncify what
+/// should the call loop do next
+#[derive(Debug)]
+pub enum OnCalledAction {
+    /// Will call the function again
+    InvokeAgain,
+    /// Will return the result of the invocation
+    Finish,
+    /// Traps with an error
+    Trap(Box<dyn std::error::Error + Send + Sync>),
+}
+
```

Comment:
Move this to WASI (up to discussion)

Reply:
see earlier comment this is needed for `asyncify` to work - it can't be moved into WASI completely as the main loop needs to change hence this was an attempt to at least keep the changes needed to a minimum in the main API.

Reply:
Can we add a new comment linking to the asyncify refactor afterwards:

```
// TODO: issue #XYZ 
```

Reply:
done